File: ASCOM.Utilities/ASCOM.Utilities.Support/EntityClasses/ConfiguredDevice.cs
using System;

namespace ASCOM.Utilities.Support
{
    /// <summary>
    /// Configured device entity class
    /// </summary>
    public class ConfiguredDevice
    {
        /// <summary>
        /// Default initialiser
        /// </summary>
        public ConfiguredDevice() { }

        /// <summary>
        /// Initialise the device name, device type, device number and ASCOM device unique ID
        /// </summary>
        /// <param name="deviceName">ASCOM device name</param>
        /// <param name="deviceType">ASCOM device type</param>
        /// <param name="deviceNumber">Alpaca API device number</param>
        /// <param name="uniqueID">ASCOM device unique ID</param>
        public ConfiguredDevice(string deviceName,string deviceType,int deviceNumber,string uniqueID)
        {
            DeviceName = deviceName;
            DeviceType = deviceType;
            DeviceNumber = deviceNumber;
            UniqueID = uniqueID;
        }

        /// <summary>
        /// ASCOM device name
        /// </summary>
        public string DeviceName { get; set; }

        /// <summary>
        /// ASCOM device type
        /// </summary>
        public string DeviceType { get; set; }

        /// <summary>
        /// Alpaca API device number
        /// </summary>
        public int DeviceNumber { get; set; }

        /// <summary>
        /// ASCOM device unique ID
        /// </summary>
        public string UniqueID { get; set; }
    }
}

File: ASCOM.Utilities/ASCOM.Utilities/AlpacaSupport/EntityClasses/ConfiguredDevice.vb
Namespace ASCOM.Utilities.Support
    ''' <summary>
    ''' Configured device entity class
    ''' </summary>
    Public Class ConfiguredDevice
        ''' <summary>
        ''' Default initialiser
        ''' </summary>
        Public Sub New()
        End Sub


        ''' <summary>
        ''' Initialise the device name, device type, device number and ASCOM device unique ID
        ''' </summary>
        ''' <paramname="deviceName">ASCOM device name</param>
        ''' <paramname="deviceType">ASCOM device type</param>
        ''' <paramname="deviceNumber">Alpaca API device number</param>
        ''' <paramname="uniqueID">ASCOM device unique ID</param>
        Public Sub New(ByVal deviceName As String, ByVal deviceType As String, ByVal deviceNumber As Integer, ByVal uniqueID As String)
            Me.DeviceName = deviceName
            Me.DeviceType = deviceType
            Me.DeviceNumber = deviceNumber
            Me.UniqueID = uniqueID
        End Sub


        ''' <summary>
        ''' ASCOM device name
        ''' </summary>
        Public Property DeviceName As String

        ''' <summary>
        ''' ASCOM device type
        ''' </summary>
        Public Property DeviceType As String

        ''' <summary>
        ''' Alpaca API device number
        ''' </summary>
        Public Property DeviceNumber As Integer

        ''' <summary>
        ''' ASCOM device unique ID
        ''' </summary>
        Public Property UniqueID As String
    End Class
End Namespace
